### DISCUSSION TITLE
Error downloading Filament generated Excel export when deployed to Vapor

### DISCUSSION THREAD

**Discussion by eacsoftware-chris** [filamentphp/filament#14194] [Help]
### Package

Actions

### Package Version

V3.2.40

### How can we help you?

I'm using the Export action to export table data to CSV and Excel. This works fine on my local environment, but once deployed to Vapor and configured to use s3, CSVs download fine but XLSX downloads fail with a 502 Bad Gateway.

I suspect the issue may be with binary responses from Vapor in general as detailed here: [https://docs.vapor.build/projects/development.html], however I don't see a way I can hook into the Filament XlsxDownloader to add the recommended:

`return $response->withHeaders([
    'X-Vapor-Base64-Encode' => 'True',
]);`

Or am I on the wrong track altogether and others have managed to get XLSX downloads to work fine on Vapor?

**Comment by bobwurtz** [comment]
I am also experiencing this problem. I switched to CSV exports and it works without issue.

**Comment by Robertbaelde** [comment]
I think the issue is because Lambda doesn't allow streamedDownload responses. I've also encountered the issue and this PR should fix it: https://github.com/filamentphp/filament/pull/14591
